pages/[id].js

```diff
@@ -11,11 +11,20 @@ export default function Detail() {
   const { id } = router.query;
   const toast = useToast();
   const { data, isLoading } = useSWR(`api/roasts/${id}`, fetcher);
-
-  if (isLoading) return <h1>... is Loading</h1>;
-
+  const { data: favorites, isLoading: favoritesLoading } = useSWR(
+    () => (session ? "api/user" : null),
+    fetcher,
+    { refreshInterval: 1000 }
+  );
+  if (isLoading || favoritesLoading) return <h1>... is Loading</h1>;
   if (!data) return <h1>Den Kaffee gibts wohl nicht ...</h1>;
 
+  function isFavorite(id) {
+    if (session) {
+      return favorites.roasts.some((roast) => roast._id === id);
+    } else return false;
+  }
+
```

Comment:
You could simpliy this to immediately get the result.
```suggestion
      const isFavorite = session && favorites.roasts.some((roast) => roast._id === id);
```